lib/Window.js

```diff
@@ -22,106 +22,219 @@ module.exports = (Item) => {
       this.selectedItem = null
     }
 
-    acceptClick (click) {
-      assert.ok(click.mouseButton === 0 || click.mouseButton === 1)
+    acceptClick (click, gamemode = 0) {
+      const { mode, slot, mouseButton } = click
+      assert.ok(
+        (mode >= 0 && mode <= 6) &&
+        (mouseButton >= 0 && mouseButton <= 8) &&
+        ((slot >= 0 && slot < this.inventoryEnd) || slot === -999),
+        'invalid operation')
+
+      switch (click.mode) {
+        case 0:
+          assert.ok(mouseButton <= 1, 'invalid operation')
+          this.mouseClick(click)
+          break
+
+        case 1:
+          assert.ok(mouseButton <= 1, 'invalid operation')
+          this.shiftClick(click)
+          break
+
+        case 2:
+          assert.ok(mouseButton <= 8, 'invalid operation')
```

Comment:
It looks like mouseButton for above 8 is possible in at least 1.19.4 (Although not vanilla) https://web.archive.org/web/20230519040131/https://wiki.vg/Protocol#Click_Container (web archive because I can't seam to access wiki.vg). So some sort of switch for newer versions would be nice at some point.